### PR TITLE
Add last modified for audit log

### PR DESCRIPTION
**Pull request by pratikmishra356**
## Problem
Describe the problem you are trying to solve here

## Solution
Provide a brief summary of your solution so that reviewers can understand your code

## Environment variable changes

What ENVs need to be added or changed

## Pre-deployment activity
Things needed to be done before deploying this change (if any)

## Post-deployment activity
Things needed to be done after deploying this change (if any)

## API changes
| Endpoint        | Method           | Request body  | Response Body |
| ------------- |:-------------:| -----:| ----------------:|
| API      | GET/POST, etc | request | response |

## Possible Issues in the future
Describe any possible issues that could occur because of this change

### CODE REVIEW

File: crates/context_aware_config/migrations/2024-04-22-122806_config_verions/down.sql [hunk 1]
```diff
@@ -1 +1,3 @@
 -- This file should undo anything in `up.sql`
+DELETE FROM public.config_versions;
+DROP TABLE public.config_versions;
```
Comment (Datron) [inline crates/context_aware_config/migrations/2024-04-22-122806_config_verions/down.sql:3]:
```suggestion
DROP TABLE IF EXISTS public.config_versions;
```

File: crates/context_aware_config/src/api/default_config/handlers.rs [hunk 2]
```diff
@@ -117,8 +131,10 @@ async fn create(
         value,
         schema,
         function_name,
-        created_by: user.get_email(),
-        created_at: Utc::now(),
+        created_by: created_by_val,
+        created_at: created_at_val,
+        last_modified_at: Utc::now().naive_utc(),
```
Comment (Datron) [inline crates/context_aware_config/src/api/default_config/handlers.rs:136]:
```suggestion
        last_modified_at: created_at_val,
```
These two should be the same right? Lets not recompute the time here

**Review by Datron — CHANGES_REQUESTED** [review]
Few small comments, can we quickly resolve them?

File: crates/context_aware_config/src/api/default_config/handlers.rs (hunk 2 above)
Comment (pratikmishra356) [reply crates/context_aware_config/src/api/default_config/handlers.rs:136]: no @Datron , created_at_val can be from already existing row, thats why recomputed the time

Comment (Datron) [reply crates/context_aware_config/src/api/default_config/handlers.rs:136]: @pratikmishra356 this is inside create function, how can created_at_val previously exist? Either way, the mark is for this row right? created_at should be recomputed and the same should be used for last_modified_at

Comment (pratikmishra356) [reply crates/context_aware_config/src/api/default_config/handlers.rs:136]: @Datron the handler name is create , but we have only one api for create or update/ the handler name is misguiding 
we create default_config struct from combination of req data and existing data , 
`let result = fetch_default_key(&key, &mut conn);`

**Review by Datron — APPROVED** [review]

**Comment by pratikmishra356** [comment]
do not merge this without letting the team know , as it contains db changes it might block cac deployment

File: crates/context_aware_config/src/api/context/handlers.rs [hunk 3]
```diff
@@ -264,11 +273,16 @@ fn replace_override_of_existing_ctx(
         override_id: new_override_id,
         ..ctx
     };
-    diesel::update(dsl::contexts)
+    let update_resp = diesel::update(dsl::contexts)
```
Comment (mahatoankitkumar) [inline crates/context_aware_config/src/api/context/handlers.rs:276]: Can we remove this code repetition?
same as line 251

**Review by mahatoankitkumar — CHANGES_REQUESTED** [review]

**Review by mahatoankitkumar — APPROVED** [review]

**Pull request merged**